qemu-psa-sp.mk

```diff
@@ -0,0 +1,143 @@
+OUT_PATH		?= $(ROOT)/out
+
+QEMU_VIRTFS_HOST_DIR = $(ROOT)
+QEMU_VIRTFS_AUTOMOUNT = y
+UBOOT ?= n
+MEASURED_BOOT			?= y
+MEASURED_BOOT_FTPM		?= n
+TS_SMM_GATEWAY			?= y
+TS_UEFI_TESTS			?= y
+# Supported values: embedded (fip is currently not supported on QEMU platform)
+SP_PACKAGING_METHOD		?= embedded
+SPMC_TESTS			?= n
+
+# Disable packages not used by this configuration
+BR2_PACKAGE_HOST_E2FSPROGS	?= n
+BR2_PACKAGE_KEYUTILS		?= n
+BR2_PACKAGE_MMC_UTILS		?= n
+BR2_PACKAGE_OPENSC		?= n
+BR2_PACKAGE_OPTEE_EXAMPLES_EXT	?= n
+BR2_PACKAGE_STRACE		?= n
+
+# Building xtest is not necessary if we don't want to run the SPMC tests
+ifneq ($(SPMC_TESTS),y)
+BR2_PACKAGE_OPTEE_TEST_EXT	?= n
+BR2_PACKAGE_LIBOPENSSL		?= n
+BR2_PACKAGE_OPENSSL		?= n
+endif
+
+# TS SP configurations
+DEFAULT_SP_CONFIG		?= default-opteesp
+SP_BLOCK_STORAGE_CONFIG	?= $(DEFAULT_SP_CONFIG)
+SP_PSA_ITS_CONFIG		?= $(DEFAULT_SP_CONFIG)
+SP_PSA_PS_CONFIG		?= $(DEFAULT_SP_CONFIG)
+SP_PSA_CRYPTO_CONFIG		?= $(DEFAULT_SP_CONFIG)
+SP_PSA_ATTESTATION_CONFIG	?= $(DEFAULT_SP_CONFIG)
+SP_SMM_GATEWAY_CONFIG		?= $(DEFAULT_SP_CONFIG)
+
+# SPMC runs at EL1
+SPMC_AT_EL ?= 1
+# Use GICv2 or GICv3
+GICV3 ?= y
+# Use TF-A's Trusted Board Boot mechanism?
+TF_A_TRUSTED_BOARD_BOOT ?= y
+
+TF_A_FLAGS ?= \
+	BL32=$(OPTEE_OS_PAGER_V2_BIN) \
+	BL33=$(BL33_BIN) \
+	PLAT=qemu \
+	SPD=spmd \
+	SPMD_SPM_AT_SEL2=0 \
+	$(TF_A_FIP_SP_FLAGS) \
+	QEMU_USE_GIC_DRIVER=$(TFA_GIC_DRIVER) \
+	QEMU_TOS_FW_CONFIG_DTS=$(ROOT)/build/qemu_v8/spmc_el1_manifest.dts \
+	SP_LAYOUT_FILE=$(TS_INSTALL_PREFIX)/sp_layout.json
+
+
+QEMU_EXTRA_ARGS ?= \
+	-cpu cortex-a57 \
+	-semihosting-config enable=on,target=native,userspace=on \
+	-d trace:'__*'
+
+include qemu_v8.mk
+include trusted-services.mk
+
+# The macros used in bl2_sp_list.dts and spmc_manifest.dts has to be passed to
+# TF-A because it handles the preprocessing of these files.
+define add-dtc-define
+DTC_CPPFLAGS+=-D$1=$(subst y,1,$(subst n,0,$($1)))
+endef
+
+ifeq ($(SP_PACKAGING_METHOD),fip)
+$(eval $(call add-dtc-define,SPMC_TESTS))
+$(eval $(call add-dtc-define,TS_SMM_GATEWAY))
+
+TF_A_EXPORTS += DTC_CPPFLAGS="$(DTC_CPPFLAGS)"
+endif
+
+OPTEE_OS_COMMON_EXTRA_FLAGS += \
+	CFG_SECURE_PARTITION=y \
+	CFG_CORE_SEL1_SPMC=y \
+	CFG_CORE_HEAP_SIZE=131072 \
+	CFG_DT=y \
+	CFG_MAP_EXT_DT_SECURE=y
+
+TS_APP_COMMON_FLAGS +=-DMM_COMM_BUFFER_ADDRESS="0x42000000"
+
+# The boot order of the SPs is determined by the order of calls here. This is
+# due to the SPMC not (yet) supporting the boot order field of the SP manifest.
+ifeq ($(SPMC_TESTS),n)
```

Comment:
`ifneq ($(SPMC_TESTS),y`)?

Reply:
The same pattern is used for the build file for FVP (`fvp-psa-sp.mk`), so it might be better to keep it for easier maintenance.

Reply:
ok